### PR TITLE
[ELF] Remove `consumeLabel` in ScriptLexer

### PR DESCRIPTION
**Pull request by hongyu-dev**
This commit removes `consumeLabel` since we can just use consume function to have the same functionalities.

### CODE REVIEW

File: lld/ELF/ScriptParser.cpp [hunk 1]
```diff
@@ -1692,11 +1692,11 @@ ScriptParser::readSymbols() {
   while (!errorCount()) {
     if (consume("}"))
       break;
-    if (consumeLabel("local")) {
+    if (consume("local:") || (consume("local") && consume(":"))) {
       v = &locals;
       continue;
     }
-    if (consumeLabel("global")) {
+    if (consume("global:") || (consume("global") && consume(":"))) {
```
Comment (MaskRay) [inline lld/ELF/ScriptParser.cpp:1699]: We should check `local:` and `global:` below at `StringRef tok = next();`

Comment (yugier) [reply lld/ELF/ScriptParser.cpp:1699]: Sorry I am confused here why we need check `local:` and `global` below `StringRef tok = next();` ?  if the lexer didn't consume `global:`, `global : `, `local:`, `local : `, or `extern`, we do not need do other things for the symbol right? Please correct me if I understand here wrong. Thank you!

Comment (MaskRay) [reply lld/ELF/ScriptParser.cpp:1699]: * Manually inline the two `consumeLabel`  calls
* Move the expanded body to below, just above `next()`

Comment (yugier) [reply lld/ELF/ScriptParser.cpp]: I committed the change but I think it's not what you were trying to say? If it is not could you please specify it more?

File: lld/ELF/ScriptParser.cpp [hunk 2]
```diff
@@ -1719,19 +1719,20 @@ ScriptParser::readSymbols() {
...
-    if (consumeLabel("global")) {
-      v = &globals;
-      continue;
-    }
 
     if (consume("extern")) {
       SmallVector<SymbolVersion, 0> ext = readVersionExtern();
       v->insert(v->end(), ext.begin(), ext.end());
     } else {
+      if (consume("local:") || (consume("local") && consume(":"))) {
```
Comment (MaskRay) [inline lld/ELF/ScriptParser.cpp:1727]: I think this introduces a bug that a symbol named `local` or `global` will not be accepted.

The "local" and "global" name check should be performed after `next()`.

Comment (yugier) [reply lld/ELF/ScriptParser.cpp]: It makes a lot of sense now thank you very much! I just updated it again.

**Review by MaskRay — APPROVED** [review]

**Pull request merged**